nonsemantic/NonSemantic.ClspvReflection.asciidoc

```diff
@@ -0,0 +1,666 @@
+
+This instruction set requires SPV_KHR_non_semantic_info.
+
+1. Introduction
+---------------
+
+This specifies the NonSemantic.ClspvReflection extended instruction set. It
+provides all the necessary instructions needed to relay the reflection
+information for https://github.com/google/clspv[clspv]-generated shader modules
+to a runtime (e.g https://github.com/kpet/clvk[clvk]). It is not expected that
+drivers implement this instruction set.
+
+Import this extended instruction set using an *OpExtInstImport*
+"NonSemantic.ClspvReflection.'<ver>'" instruction, where '<ver>' indicates the
+version.
+
+2. Version
+----------
+
+This extended instruction set is versioned. The version is specified via the
+import string. It must be an integer between 1 and the version specified at the
+beginning of this instruction set. Differences between the version are
+specified below.
+
+3. Binary Form
+---------------
+
+The return type for all instructions must be *OpTypeVoid*.
+
+[cols="3"]
+|=====
+3+|[[Kernel]]*Kernel* +
+ +
+Declares a shader entry-point generated by clspv. +
+ +
+'Kernel' must be an *OpFunction* that is declared as a *GLCompute* entry-point. +
+ +
+'Name' must be an *OpString* specifying the name of the entry-point. Name must 
+match an entry-point name for 'Kernel'.
+1+| 1 | '<id>' +
+'Kernel' |
+'<id>' +
+'Name'
+|=====
+
+[cols="6"]
+|=====
+6+|[[ArgumentInfo]]*ArgumentInfo* +
+ +
+The operands in this instruction are suitable for return from clGetKernelArgInfo. +
+ +
+'Name' must be an *OpString* specifying the name of the kernel argument. +
+ +
+'TypeName' must be an *OpString* specifying the type of the kernel argument. +
+ +
+'AddressQualifier' must be a 32-bit unsigned integer *OpConstant* specifying the address space
+enum value of the kernel argument. +
+ +
+'AccessQualifier' must be a 32-bit unsigned integer *OpConstant* specifying the access qualifier
+enum value of the kernel argument. +
+ +
+'TypeQualifier' must be a 32-bit unsigned integer *OpConstant* specifying the type qualifier
+enum value of the kernel argument.
+1+| 2 | '<id>' +
+'Name' |
+Optional +
+'<id>' +
+'TypeName' |
+Optional +
+'<id>' +
+'AddressQualifier' |
+Optional +
+'<id>' +
+'AccessQualifier' |
+Optional +
+'<id>' +
+'TypeQualifier'
+|=====
+
+[cols="6"]
+|=====
+6+|[[ArgumentStorageBuffer]]*ArgumentStorageBuffer* +
+ +
+Declares a *StorageBuffer* argument for 'Kernel'. The Vulkan descriptor type should use
+*VK_DESCRIPTOR_TYPE_STORAGE_BUFFER*. +
+ +
+'Kernel' must be a <<Kernel,*Kernel*>> extended instruction from the same import. +
+ +
+'Ordinal' must be a 32-bit unsigned integer *OpConstant* specifying the argument ordinal. +
+ +
+'DescriptorSet' must be a 32-bit unsigned integer *OpConstant* specifying the descriptor set. +
+ +
+'Binding' must be a 32-bit unsigned integer *OpConstant* specifying the binding. +
+ +
+'ArgInfo' must be an <<ArgumentInfo,*ArgumentInfo*>> extended instruction from the same import.
```

Comment:
It might be nice to note early in this extension that there are no forward references among these instructions. So both the referenced Kernel and ArgumentInfo should have appeared earlier in the instruction stream.